kamu-core/src/domain/repos/metadata_chain_comparator.rs

```diff
@@ -23,12 +24,35 @@ impl MetadataChainComparator {
         lhs_head: &Multihash,
         rhs_chain: &dyn MetadataChain,
         rhs_head: Option<&Multihash>,
+        listener: &dyn CompareChainsListener,
     ) -> Result<CompareChainsResult, CompareChainsError> {
         // When source and destination point to the same block, chains are equal, no further scanning required
         if Some(&lhs_head) == rhs_head.as_ref() {
             return Ok(CompareChainsResult::Equal);
         }
 
+        let lhs_chain = MetadataChainWithStats::new(
```

Comment:
Not too happy with this... I suspect it's not the last time we need some introspection into what's happening in the metadata chain, so we might support some listener interface directly in `MetadataChain` trait in the future.